src/Concordium/Client/Runner.hs

```diff
@@ -3445,81 +3448,391 @@ processIdentityShowCmd action backend =
 processLegacyCmd :: LegacyCmd -> Backend -> IO ()
 processLegacyCmd action backend =
   case action of
+    -- ↓ FIXME: This should be verified in particular, 
+    --          I am not sure how to test it.
     SendTransaction fname nid -> do
       source <- handleReadFile BSL.readFile fname
       t <- withClient backend $ processTransaction source nid
       putStrLn $ "Transaction sent to the baker. Its hash is " ++
         show (getBlockItemHash t)
-    GetConsensusInfo -> withClient backend $ getConsensusStatus >>= printJSON . fmap grpcResponseVal
-    GetBlockInfo every block -> withClient backend $ withBestBlockHash block getBlockInfo >>= if every then loop . fmap grpcResponseVal else printJSON . fmap grpcResponseVal
-    GetBlockSummary block -> withClient backend $ withBestBlockHash block getBlockSummary >>= printJSON . fmap grpcResponseVal
-    GetBlocksAtHeight height gen restr -> withClient backend $ getBlocksAtHeight height gen restr >>= printJSON . fmap grpcResponseVal
-    GetAccountList block -> withClient backend $ withBestBlockHash block getAccountList >>= printJSON . fmap grpcResponseVal
-    GetInstances block -> withClient backend $ withBestBlockHash block getInstances >>= printJSON . fmap grpcResponseVal
-    GetTransactionStatus txhash -> withClient backend $ getTransactionStatus txhash >>= printJSON . fmap grpcResponseVal
-    GetTransactionStatusInBlock txhash block -> withClient backend $ getTransactionStatusInBlock txhash block >>= printJSON . fmap grpcResponseVal
-    GetAccountInfo account block ->
-      withClient backend $ withBestBlockHash block (getAccountInfo account) >>= printJSON . fmap grpcResponseVal
-    GetAccountNonFinalized account ->
-      withClient backend $ getAccountNonFinalizedTransactions account >>= printJSON . fmap grpcResponseVal
-    GetNextAccountNonce account ->
-      withClient backend $ getNextAccountNonce account >>= printJSON . fmap grpcResponseVal
-    GetInstanceInfo account block ->
-      withClient backend $ withBestBlockHash block (getInstanceInfo account) >>= printJSON . fmap grpcResponseVal
+    GetConsensusInfo ->
+      withClient backend $
+        getConsensusInfoV2 >>=
+        printResponseValueAsJSON'
+    GetBlockInfo every block ->
+      withClient backend $
+        readOrFailM Best block >>=
+        printBlockInfos every
+    GetBlockSummary block ->
```

Comment:
In light of the fact that the the node is missing an endpoint, please do the following

- split this command into the 4 ones
- GetBlockEvents
- GetBlockSpecialEvents
- GetChainParameters
- GetBlockPendingUpdates

That'll also be faster to use. Then we are not blocked by the node changes.

Reply:
Should `GetBlockPendingUpdates` output a JSON representation of the raw response or the response converted into a `PendingUpdates` instance containing the individual queues (lifting the helper functions I already implemented)?

I guess the former since it's a `raw` command?

Reply:
Whatever is easier, really.